### PR TITLE
handle notfound and notpermitted error in Smb::getDirectoryContent

### PR DESCRIPTION
**Pull request by icewind1991**
Signed-off-by: Robin Appelman <robin@icewind.nl>

### CODE REVIEW

File: apps/files_external/lib/Lib/Storage/SMB.php [hunk 1]
```diff
@@ -620,7 +620,13 @@ public function opendir($path) {
 	}
 
 	public function getDirectoryContent($directory): \Traversable {
-		$files = $this->getFolderContents($directory);
+		try {
+			$files = $this->getFolderContents($directory);
```
Comment (PVince81) [inline apps/files_external/lib/Lib/Storage/SMB.php:624]: I was about to ask whether we need this in `opendir` as well and noticed that we already catch those there

so probably this change is a step in the right direction

Comment (PVince81) [reply apps/files_external/lib/Lib/Storage/SMB.php:624]: hmm, but `opendir` returns false, which signifies an error

should we throw as a different Nextcloud-specific error, or return null ? in a generator ? :thinking:

Comment (icewind1991) [reply apps/files_external/lib/Lib/Storage/SMB.php:624]: The default implementation in `Storage\Common` currently has the same behavior when failing to open the directory

**Review by come-nc — APPROVED** [review]

**Review by PVince81 — APPROVED** [review]
:+1:

**Pull request merged**